lib/types.ts

```diff
@@ -68,3 +68,12 @@ export type ListDisplayMode = 'expanded' | 'comfy' | 'condensed';
 export type SortType = 'alphabetical' | 'creationDate' | 'modificationDate';
 export type Theme = 'system' | 'light' | 'dark';
 export type TranslatableString = string;
+
+///////////////////////////////////////
+// Language and Platform
+///////////////////////////////////////
+
+export type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
```

Comment:
I feel like these need some explanation as to what they are.

Reply:
can do and will add. I feel like `XOR` should be in the language. it's here because the `|` is only really useful when we have a property on which to discriminate alternative types.

```ts
type Colors = { name: 'red', value: 0xff000 } | { name: 'green', value: 0x00ff00 };
```

This fails if our types our different

```ts
type Size = { radius: number } | { width: number; height: number };
```

It _should_ be able to do this but it can't. When we don't have that property we have to rely on more advanced typing. In this case it's a mapped type called `XOR` which does exactly what `|` seems like it should do. `Without` is part of it and gets the difference of keys between them. Here I've used `XOR` because `MouseEvent` and `TouchEvent` don't have the discriminating property.